mkosi/__init__.py

```diff
@@ -4010,6 +4010,7 @@ def run_shell(args: Args, config: Config) -> None:
                     network=True,
                     devices=True,
                     options=["--bind", fname, workdir(fname)],
+                    setup=become_root_cmd(),
```

Comment:
As far as I understand the idea is to be able to run `mkosi boot` without `sudo` (and with the latest changes it works). Though it breaks when things like `--register=yes/Register=` are added because `mkosi` bails out with
```sh
Container registration requires root privileges
```
even though it should have no problem with that in theory. (I'm not sure how important it is though. It's just something I noticed when I poked mkosi in the context of HACKING.md)

Reply:
It's of course possible to get it around with
```sh
mkosi boot --register=yes
```